plugins/inputs/socketstat/socketstat.go

```diff
@@ -0,0 +1,219 @@
+package socketstat
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Socketstat is a telegraf plugin to gather indicators from established connections, using iproute2's  `ss` command.
+type Socketstat struct {
+	BeginsWithBlank *regexp.Regexp
+	cmdName         string
+	Lister          socketLister
+	Log             telegraf.Logger
+	Measurement     string
+	SocketProto     []string
+	Timeout         config.Duration
+	ValidValues     *regexp.Regexp
```

Comment:
As far as I can see, only two of those variables (`Timeout` and `SocketProto`) are actually config options, so please add toml-tags and make all possible values "unexported":
```suggestion
	SocketProto     []string `toml:"socket_proto"`
	Timeout         config.Duration `toml:"timeout"`
	Log             telegraf.Logger `toml:"-"`

	beginsWithBlank *regexp.Regexp
	validValues     *regexp.Regexp
	cmdName         string
	lister          socketLister
	measurement     string
```

Reply:
Done.